File: src/org/apache/logging/log4j/LogManager.java
package org.apache.logging.log4j;

import kotlin.reflect.KClass;

import java.util.HashMap;
import java.util.HashSet;

/**
 * the main logging manager, which should be used
 * */
public class LogManager {

    private static final HashSet<String> disabled = new HashSet<>();

    public static boolean isEnabled(LoggerImpl logger){
        return !disabled.contains(logger.getPrefix());
    }

    public static void disableLogger(String logger){
        disabled.add(logger);
    }

    public static void enableLogger(String logger){
        disabled.remove(logger);
    }

    private static final LoggerImpl logger = new LoggerImpl(null);
    private static final HashMap<String, LoggerImpl> loggers = new HashMap<>();

    public static LoggerImpl getLogger(){
        return logger;
    }

    public static LoggerImpl getLogger(Class<?> clazz){
        return getLogger(clazz.getSimpleName());
    }

    public static LoggerImpl getLogger(KClass<?> clazz){
        return getLogger(clazz.getSimpleName());
    }

    public static LoggerImpl getLogger(String name){
        if(name == null) return logger;
        LoggerImpl logger = loggers.get(name);
        if(logger == null){
            logger = new LoggerImpl(name);
            loggers.put(name, logger);
        }
        return logger;
    }

}

File: src/org/apache/logging/log4j/LogManager.kt
package org.apache.logging.log4j

import kotlin.reflect.KClass

/**
 * the main logging manager, which should be used all over the engine;
 * @see [me.anno.ui.debug.ConsoleOutputPanel]
 */
object LogManager {

    @JvmStatic
    private val logLevels = HashMap<String?, Level?>()

    @JvmStatic
    private var defaultLevel = Level.INFO

    @JvmStatic
    fun isEnabled(logger: LoggerImpl, level: Level): Boolean {
        val value = logLevels[logger.name] ?: defaultLevel
        return level <= value
    }

    @JvmStatic
    fun isEnabled(logger: LoggerImpl, level: Int): Boolean {
        val value = logLevels[logger.name] ?: defaultLevel
        return level <= value.value
    }

    @JvmStatic
    fun disableLogger(logger: String?) {
        logLevels[logger] = Level.OFF
    }

    @JvmStatic
    fun disableLoggers(loggers: String) {
        for (logger in loggers.split(',')) {
            disableLogger(logger)
        }
    }

    @JvmStatic
    fun disableInfoLogs(loggers: String) {
        for (logger in loggers.split(',')) {
            logLevels[logger] = Level.WARN
        }
    }

    @JvmStatic
    @Suppress("unused")
    fun enableLogger(logger: String?) {
        define(logger, defaultLevel)
    }

    @JvmStatic
    @Suppress("unused")
    fun enableLogger(logger: LoggerImpl) {
        enableLogger(logger.name)
    }

    @JvmStatic
    fun define(logger: String?, level: Level) {
        logLevels[logger] = level
    }

    @JvmStatic
    fun default(level: Level) {
        defaultLevel = level
    }

    @JvmStatic
    fun logAll() {
        default(Level.ALL)
    }

    @JvmStatic
    private val loggers = HashMap<String, LoggerImpl>(256)

    @JvmStatic
    fun getLogger(clazz: Class<*>): LoggerImpl {
        return getLogger(clazz.simpleName)
    }

    @JvmStatic
    fun getLogger(clazz: KClass<*>): LoggerImpl {
        return getLogger(clazz.simpleName)
    }

    @JvmStatic
    fun getLogger(name: String?): LoggerImpl {
        val nameI = name ?: ""
        return loggers.getOrPut(nameI) { LoggerImpl(nameI) }
    }
}